File: test/aidbox_sdk/snapshots/typescript/hl7-fhir-r4-core/RiskAssessment.ts
import { Annotation } from "./Annotation";
import { Period } from "./Period";
import { CodeableConcept } from "./CodeableConcept";
import { Range } from "./Range";
import { DomainResource } from "./DomainResource";
import { Reference } from "./Reference";
import { Identifier } from "./Identifier";
import { BackboneElement } from "./BackboneElement";

export type RiskAssessmentPrediction = BackboneElement & {
    relativeRisk?: number;
    when?: Range | Period;
    outcome?: CodeableConcept;
    probability?: Range | number;
    rationale?: string;
    qualitativeRisk?: CodeableConcept;
};

export type RiskAssessment = DomainResource & {
    parent?: Reference;
    encounter?: Reference;
    prediction?: RiskAssessmentPrediction[];
    method?: CodeableConcept;
    basis?: Reference[];
    reasonCode?: CodeableConcept[];
    mitigation?: string;
    note?: Annotation[];
    status: string;
    condition?: Reference;
    code?: CodeableConcept;
    identifier?: Identifier[];
    basedOn?: Reference;
    subject: Reference;
    occurrence?: Period | string;
    performer?: Reference;
    reasonReference?: Reference[];
};

File: test/aidbox_sdk/snapshots/python/hl7-fhir-r4-core/RiskAssessment.py
from typing import Optional, List
from pydantic import *
from ..base import *

class RiskAssessment_Prediction(BackboneElement):
    relative_risk: Optional[float] = None
    when_range: Optional[Range] = None
    outcome: Optional[CodeableConcept] = None
    when_period: Optional[Period] = None
    rationale: Optional[str] = None
    probability_range: Optional[Range] = None
    qualitative_risk: Optional[CodeableConcept] = None
    probability_decimal: Optional[float] = None

class RiskAssessment(DomainResource):
    parent: Optional[Reference] = None
    encounter: Optional[Reference] = None
    prediction: Optional[List[RiskAssessment_Prediction]] = None
    method: Optional[CodeableConcept] = None
    basis: Optional[List[Reference]] = None
    reason_code: Optional[List[CodeableConcept]] = None
    mitigation: Optional[str] = None
    note: Optional[List[Annotation]] = None
    occurrence_period: Optional[Period] = None
    status: str
    condition: Optional[Reference] = None
    code: Optional[CodeableConcept] = None
    identifier: Optional[List[Identifier]] = None
    based_on: Optional[Reference] = None
    occurrence_date_time: Optional[str] = None
    subject: Reference
    performer: Optional[Reference] = None
    reason_reference: Optional[List[Reference]] = None